File: inf_04_2025_01_02_SG/mobilna/app/src/main/java/com/example/mobilna/MainActivity.java
package com.example.mobilna;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import java.util.concurrent.atomic.AtomicBoolean;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });


        EditText nrPraniaEditText = findViewById(R.id.nrPraniaEditText);
        Button zatwierdzButton = findViewById(R.id.zatwierdzButton);
        TextView nrPraniaText = findViewById(R.id.numer_prania);


        Button wlaczWylaczButton = findViewById(R.id.wlaczWylaczButton);
        TextView odkurzaczWlaczony = findViewById(R.id.odkurzaczWlaczonyWylaczony);
        AtomicBoolean wlaczony = new AtomicBoolean(true);

        zatwierdzButton.setOnClickListener(v -> {
            int nrPrania = Integer.parseInt(nrPraniaEditText.getText().toString());

            if(nrPrania >= 1 && nrPrania <= 12){
                nrPraniaText.setText("Numer prania: " + nrPrania);
            }
        });

        wlaczWylaczButton.setOnClickListener(v -> {
            wlaczony.set(!wlaczony.get());
            if(!wlaczony.get()){
                wlaczWylaczButton.setText("Wyłącz");
                odkurzaczWlaczony.setText("Odkurzacz Włączony");
            }else{
                wlaczWylaczButton.setText("Włącz");
                odkurzaczWlaczony.setText("Odkurzacz wyłączony");
            }
        });
    }
}

File: 2023_01_02/mobilna/app/src/main/java/com/example/mobilna/MainActivity.kt
package com.example.mobilna

import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.EditText
import android.widget.ListView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        val nowyElementText = findViewById<EditText>(R.id.nowyElementText)
        val dodajButton = findViewById<Button>(R.id.dodajPrzycisk);
        val lista = findViewById<ListView>(R.id.lista);

        var elementy = arrayListOf(
            "Zakupy: chleb, masło, ser",
            "Do zrobienia: obiad, umyć podłogi",
            "weekend: kino, spacer z psem")
        val adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, elementy);

        lista.adapter = adapter

        dodajButton.setOnClickListener {
            elementy.add(nowyElementText.text.toString())
            adapter.notifyDataSetChanged()
            nowyElementText.text.clear()
        }
    }
}